mongodbatlas/resource_mongodbatlas_database_user.go

```diff
@@ -115,8 +120,9 @@ func resourceMongoDBAtlasDatabaseUserRead(d *schema.ResourceData, meta interface
 			authDatabaseName = d.Get("auth_database_name").(string)
 		}
 	}
+	dbUsername := url.QueryEscape(username)
```

Comment:
Perhaps a more informative name?  username_escaped or what ever best fits standard go naming?

Also I'm a bit worried about doing this across the board as we've been able to GET and DELETE with special characters in db names before (like a dash).  Let's make sure this doesn't cause any side impacts by being sure we test that thoroughly please.

Reply:
Sure no problem, I have tested many times with testacc and all OK when creating, getting and deleting but will do more testing to be sure of course